### PR TITLE
Current ride

### PR DESCRIPTION
**Pull request by tyang98**
### Summary 

This pull request is the first step towards implementing current ride

- [x] Current ride Card
- [x] Fetch current ride from backend.
- [x] RidesProvider documentation- 
- [x] Connect driver to ride pages

### Test Plan 

Tested on an emulator. 

### Notes 

- Need to fix overflow for very long location names

### CODE REVIEW

File: lib/models/Ride.dart [hunk 1]
```diff
@@ -548,3 +569,198 @@ class RecurringRidesGenerator {
...
+        ),
+      ),
+    )
+  ]);
+}
+
+Widget arrivedRide(context, Ride ride) {
+  String startLocation = ride.startLocation;
+  return Row(children: <Widget>[
+    RichText(
```
Comment (helenxuyang) [inline lib/models/Ride.dart:601]: Expanded around here for overflow and it should be good

**Review by helenxuyang — APPROVED** [review]
Overall looks great, tested on my device and it works well except for the one missing Expanded--other suggestions are just minor ones about simplifying/organizing code!

File: lib/models/Driver.dart [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+      this.photoLink});
+
+  //Creates a driver from JSON representation.
+  factory Driver.fromJson(Map<String, dynamic> json) {
+    return Driver(
+        id: json['id'],
+        firstName: json['firstName'],
+        lastName: json['lastName'],
+        phoneNumber: json['phoneNumber'],
+        photoLink: json['photoLink']);
```
[the remark this answers is not in the document]
Comment (tyang98) [reply lib/models/Driver.dart:36]: fixed

File: lib/models/Driver.dart [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  //The first name of a driver
+  final String firstName;
+
+  //The last name of a driver
+  final String lastName;
+
+  //The phoneNumber of a driver
+  final String phoneNumber;
+
+  //The photoLink of a driver
```
[the remark this answers is not in the document]
Comment (tyang98) [reply lib/models/Driver.dart:17]: done

File: lib/models/Driver.dart [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+  //The id of a driver
+  final String id;
+
+  //The first name of a driver
+  final String firstName;
+
+  //The last name of a driver
+  final String lastName;
+
+  //The phoneNumber of a driver
```
[the remark this answers is not in the document]
Comment (tyang98) [reply lib/models/Driver.dart:14]: fixed

File: lib/models/Ride.dart [hunk 5]
```diff
@@ -138,7 +154,8 @@ class Ride {
 
   //Widget displaying a custom built card with information about a ride's start location and start time.
   //[isIcon] determines whether the card needs an icon.
-  Widget buildLocationsCard(context, bool isIcon) {
+  Widget buildLocationsCard(context, bool isIcon, bool pickUp,
+      bool isStartLocation, bool isStartAddress, bool isStartTime) {
```
[the remark this answers is not in the document]
Comment (tyang98) [reply lib/models/Ride.dart:158]: fixed

File: lib/pages/Current.dart [hunk 6]
```diff
@@ -111,16 +109,30 @@ class RecurringRide extends StatefulWidget {
 }
 
 class _RecurringRideState extends State<RecurringRide> {
```
[the remark this answers is not in the document]
Comment (tyang98) [reply lib/pages/Current.dart:111]: fixed

File: lib/providers/RidesProvider.dart [hunk 7]
```diff
@@ -10,8 +10,9 @@ import 'package:flutter/material.dart';
 import 'package:provider/provider.dart';
 import 'AuthProvider.dart';
 
-//Manage the state of rides with ChangeNotifier.
+/// Manage the state of rides with ChangeNotifier.
 class RidesProvider with ChangeNotifier {
+  Ride currentRide = Ride();
```
[the remark this answers is not in the document]
Comment (tyang98) [reply lib/providers/RidesProvider.dart:15]: Flutter tells us not to explicitly initialize variables to null

File: lib/providers/RidesProvider.dart [hunk 8]
```diff
@@ -26,21 +27,59 @@ class RidesProvider with ChangeNotifier {
...
+      print(response.body);
+      currentRide = ride;
+      print(currentRide);
+    } else {
+      throw Exception('Failed to load current ride.');
+    }
+  }
+
+  /// Fetches a list of have already occurred from the backend by using the baseUrl of [config] and rider id from [authProvider].
+  /// Past rides are sorted and displayed in order of their initial start time.
```
[the remark this answers is not in the document]
Comment (tyang98) [reply lib/providers/RidesProvider.dart:82]: fixed

File: lib/providers/RidesProvider.dart [hunk 9]
```diff
@@ -70,8 +110,10 @@ class RidesProvider with ChangeNotifier {
     }
   }
 
-  //Creates a ride in the backend by an HTTP post request with the fields:
-  //[startLocation], [endLocation], [startTime], and [endTime].
+  /// Creates a ride in the backend by an HTTP POST request with the following fields:
+  /// the location id if [startLocation] or [endLocation] is an already existing location or
+  /// just the location name if it is a new location (not in backend yet) and
```
[the remark this answers is not in the document]
Comment (tyang98) [reply lib/providers/RidesProvider.dart:115]: This is going to be rewritten in the ride flow PR

File: lib/models/Ride.dart (hunk 1 above)
Comment (tyang98) [reply lib/models/Ride.dart:601]: fixed

File: lib/models/Ride.dart [hunk 10]
```diff
@@ -548,3 +569,198 @@ class RecurringRidesGenerator {
...
+          new TextSpan(
+              text: 'complete!',
+              style: new TextStyle(fontWeight: FontWeight.bold)),
+        ],
+      ),
+    ))
+  ]);
+}
+
+Widget currentCardInstruction(context, String status, Ride ride) {
```
[the remark this answers is not in the document]
Comment (tyang98) [reply lib/models/Ride.dart:659]: sounds like a good idea!

**Pull request merged**